src/api/hooks/useGetANS.ts

```diff
@@ -0,0 +1,65 @@
+import {useEffect, useState} from "react";
+import {NetworkName} from "../../constants";
+import {useGlobalState} from "../../GlobalState";
+
+function getFetchNameUrl(network: NetworkName, address: string) {
+  switch (network) {
+    case "mainnet":
+      return `https://www.aptosnames.com/api/mainnet/v1/name/${address}`;
+    case "testnet":
+      return `https://www.aptosnames.com/api/testnet/v1/name/${address}`;
+    default:
+      return undefined;
+  }
+}
+
+export function useGetNameFromAddress(address: string) {
+  const [state, _] = useGlobalState();
+  const [name, setName] = useState<string>();
+  const url = getFetchNameUrl(state.network_name, address);
+
+  useEffect(() => {
+    if (url !== undefined) {
+      const fetchData = async () => {
+        const response = await fetch(url);
+        const {name} = await response.json();
+        setName(name);
+      };
+
+      fetchData();
+    }
+  }, [address, state]);
+
+  return name;
+}
+
+function getFetchAddressUrl(network: NetworkName, name: string) {
+  switch (network) {
+    case "mainnet":
+      return `https://www.aptosnames.com/api/mainnet/v1/address/${name}`;
+    case "testnet":
+      return `https://www.aptosnames.com/api/testnet/v1/address/${name}`;
+    default:
+      return undefined;
+  }
+}
+
+export function useGetAddressFromName(name: string) {
```

Comment:
is it being used anywhere...?

Reply:
Right now not yet. I'll work on another PR that enables searching names in the search bar. It'll be used soon.